nixos/modules/programs/rush.nix

```
@@ -0,0 +1,127 @@
+
+  indent =
+    lines:
+    lib.pipe lines [
+      (lib.splitString "\n")
+      (builtins.filter (line: line != ""))
+      (map (line: "  " + line))
+      (builtins.concatStringsSep "\n")
+    ];
+in
+{
+  meta.maintainers = with lib.maintainers; [
+    c4f3z1n
+  ];
+
+  options.programs.rush = with lib.types; {
+    enable = lib.mkEnableOption "Whether to enable Restricted User Shell." // {
+      default = opt.global.isDefined || opt.rules.isDefined;
+    };
+
+    package = lib.mkPackageOption pkgs "rush" { } // {
+      type = shellPackage;
+    };
+
+    global = lib.mkOption {
+      type = lines;
+      description = "The `global` statement defines global settings.";
+    };
+
+    rules = lib.mkOption {
+      type = attrsOf lines;
+
+      description = ''
+        The rule statement configures a GNU rush rule. This is a block statement, which means that all
+        statements located between it and the next rule statement (or end of file, whichever occurs first)
+        modify the definition of that rule.
+      '';
+    };
+
+    shell = lib.mkOption {
+      readOnly = true;
+      type = either shellPackage path;
+
+      description = ''
+        The resolved shell path that users can inherit to set `rush` as their login shell.
+        This is a convenience option for use in user definitions. Example:
+          `users.users.alice = { inherit (config.programs.rush) shell; ... };`
+      '';
+    };
+
+    wrap = lib.mkOption {
+      type = bool;
+      default = config.security.enableWrappers;
+      defaultText = lib.literalExpression "config.security.enableWrappers";
+
+      description = ''
+        Whether to wrap the `rush` binary with a SUID-enabled wrapper.
+        This is required if {option}`security.enableWrappers` is enabled in your configuration.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
+      (lib.mkIf (!lib.versionAtLeast cfg.package.version "2.0") {
+        warnings = lib.optional (!builtins.hasAttr "rush.rc" config.environment.etc) ''
+          The selected version of GNU Rush (${cfg.package.version}) is older than 2.0 and does not support configuration via Nix.
+
+          A legacy `/etc/rush.rc` file is required for versions < 2.0, but this file will **not** be created automatically by this module.
+
+          To remove this warning:
+            - Either upgrade to GNU Rush 2.0 or newer, **or**
+            - Manually provide a valid `/etc/rush.rc` file via `environment.etc."rush.rc"` in your configuration.
+        '';
+      })
+
+      (lib.mkIf (lib.versionAtLeast cfg.package.version "2.0") {
+        environment.etc."rush.rc".text =
+          lib.pipe
+            [
+              "# This file was created by the module `programs.rush`;"
+              "rush 2.0"
+              (lib.optionalString (opt.global.isDefined && cfg.global != "") "global\n${indent cfg.global}")
+              (lib.optionals (opt.rules.isDefined && cfg.rules != { }) (
```

```suggestion
              (lib.optionalString (cfg.global != "") "global\n${indent cfg.global}")
              (lib.optionals (cfg.rules != { }) (
```
If these values are not empty, they are 100% defined, that check was unnecessary.

---

(and since there's no more `opt` in use, remove it from the top of the file as well)